qiskit_experiments/library/characterization/analysis/ramsey_xy_analysis.py

```diff
@@ -436,65 +501,98 @@ def _generate_fit_guesses(
         Returns:
             List of fit options that are passed to the fitter function.
         """
-        # Compute offset guess
-        user_opt.p0.set_if_empty(
-            offset=np.mean(curve_data.y),
-            f_err=0.0,
-        )
         user_opt.bounds.set_if_empty(
-            offset=(-1, 1),
-            amp=(0, 1),
-            c2_pos=(0, np.inf),
-            c2_neg=(-np.inf, 0),
+            c2_pos=(0, np.inf), c2_neg=(-np.inf, 0)
         )
-        est_offs = user_opt.p0["offset"]
+        user_opt.p0.set_if_empty(
+            c1_pos=0, c2_pos=1, c3_pos=0, c1_neg=0, c2_neg=-1, c3_neg=0, f_err=0
+        )
+        return user_opt
 
-        # Compute amplitude guess
-        amps = np.zeros(0)
-        for direction in ("pos", "neg"):
-            ram_x_off = curve_data.get_subset_of(f"X{direction}").y - est_offs
-            ram_y_off = curve_data.get_subset_of(f"Y{direction}").y - est_offs
-            amps = np.concatenate([amps, np.sqrt(ram_x_off**2 + ram_y_off**2)])
-        user_opt.p0.set_if_empty(amp=np.median(amps))
-        est_a = user_opt.p0["amp"]
-        const = 2 * np.pi * user_opt.p0["ts"]
-
-        # Compute polynomial coefficients
+    def _create_figures(
```

Comment:
We don't need to change anything here, but it makes me wonder how far off we are from able to specify these plots declaratively in the analysis options without code. I think it is mainly the XY "fit" data that holds it back because that is not in the scatter table? Otherwise, all the data is in the scatter table and we just need to say which categories to query and what labels and styles to put on them and which canvases to put them on.

Reply:
Yes, exactly. This is needed to deduce raw XY plot. But user can also draw figures more flexibly by completely overwriting this if plot template is not sufficient.